src/components/FoodCard/index.js

```diff
@@ -0,0 +1,86 @@
+import React from "react";
+import {
+    StyledCard, CardContent, StyledCardMedia, TypographyTitle, 
+    TypographyDescription, TypographyValue, RequestQuantityWrapper, 
+    AddRequestButton, FlexDiv, TypographyButtonText
+} from "./styled";
+import { Typography } from "@material-ui/core";
+import { removeAmount } from '../../actions'
+import { connect } from 'react-redux'
+
+export const FoodCard = (props) => {
+
+    const onClickAddRequestButton = () => {
+        if (props.foodInfo.amount === 0) {
+            props.showPopUpAddCart(props.foodInfo.id)
+        } else {
+            props.removeAmount(props.foodInfo.id)
+        }
+    }
+
+    const showRequestQuantity = props.foodInfo.amount > 0 ? (
+        <RequestQuantityWrapper>
+            <Typography
+                color="primary"
+                component="p"
+                variant="caption"
+                align="center"
+            >
+                {props.foodInfo.amount}
+            </Typography >
+        </RequestQuantityWrapper>
+    ) : (<div></div>)
```

Comment:
Aqui pra evitar poluir a árvore do DOM (ou VDOM) com um div inútil, usem o `null`